seml/__main__.py

```diff
@@ -0,0 +1,678 @@
+        typer.Option(
+            '-d',
+            '--debug',
+            help="Run a single interactive experiment without Sacred observers and with post-mortem debugging. "
+                 "Implies `--verbose --num-exps 1 --post-mortem --output-to-console`.",
+        ),
+    ] = False,
+    debug_server: Annotated[
+        bool,
+        typer.Option(
+            '-ds',
+            '--debug-server',
+            help="Run the experiment with a debug server, to which you can remotely connect with e.g. VS Code. "
+                 "Implies `--debug`.",
+        ),
+    ] = False,
+    local: Annotated[
+        bool,
+        typer.Option(
+            '-l',
+            '--local',
+            help="Run the experiment locally instead of on a Slurm cluster.",
+        ),
+    ] = False,
+    no_worker: Annotated[
+        bool,
+        typer.Option(
+            '-nw',
+            '--no-worker',
+            help="Do not launch a local worker after setting experiments' state to PENDING.",
+        ),
+    ] = False,
+    num_exps: NumExperimentsAnnotation = 0,
+    file_output: FileOutputAnnotation = True,
+    steal_slurm: StealSlurmAnnotation = False,
+    post_mortem: PostMortemAnnotation = False,
+    output_to_console: OutputToConsoleAnnotation = False,
+    worker_gpus: WorkerGPUsAnnotation = None,
+    worker_cpus: WorkerCPUsAnnotation = None,
+    worker_env: WorkerEnvAnnotation = None,
+):
+    """
+    Fetch staged experiments from the database and run them (by default via Slurm).
+    """
+    assert ctx.obj['collection'], "Collection name must be passed to this command."
+    start_experiments(
+        ctx.obj['collection'],
+        local=local,
+        sacred_id=sacred_id,
+        batch_id=batch_id,
+        filter_dict=filter_dict,
+        num_exps=num_exps,
+        post_mortem=post_mortem,
+        debug=debug,
+        debug_server=debug_server,
+        output_to_console=output_to_console,
+        no_file_output=not file_output,
+        steal_slurm=steal_slurm,
+        no_worker=no_worker,
+        set_to_pending=True,
+        worker_gpus=worker_gpus,
+        worker_cpus=worker_cpus,
+        worker_environment_vars=worker_env,
+    )
+
+
+@app.command("launch-worker")
+def launch_worker_command(
+    ctx: typer.Context,
+    num_exps: NumExperimentsAnnotation = 0,
+    file_output: FileOutputAnnotation = True,
+    steal_slurm: StealSlurmAnnotation = False,
+    post_mortem: PostMortemAnnotation = False,
+    output_to_console: OutputToConsoleAnnotation = False,
+    worker_gpus: WorkerGPUsAnnotation = None,
+    worker_cpus: WorkerCPUsAnnotation = None,
+    worker_env: WorkerEnvAnnotation = None,
+    sacred_id: SacredIdAnnotation = None,
+    filter_dict: FilterDictAnnotation = None,
+    batch_id: BatchIdAnnotation = None,
+):
+    """
+    Launch a local worker that runs PENDING jobs.
+    """
+    assert ctx.obj['collection'], "Collection name must be passed to this command."
+    start_experiments(
```

Comment:
I also recognized this but decided to adopt the previous solution of having different wrappers around the same function with different defaults. Technically we could use different functions but since only 2 parameters are changed, this is also not ideal? We could also potentially overload the start function?

Reply:
IMO the main thing is that now `seml collection start` accepts flags that only make sense when `--local` is passed. Our CLI should at least warn the user about that these flags will not have the desired effect, the best would be to throw errors. 

Looking at `start.py`'s `start_experiments` it seems like maybe we should anyway disentangle the function into some version that adds to slurm and some that starts locally, with different signatures.